src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/SubscriptionsController.cs

```diff
@@ -209,6 +209,18 @@ public virtual async Task<IActionResult> UpdateSubscription(Guid id, [FromBody]
 
             if (doUpdate)
             {
+                Data.Models.Subscription equivalentSubscription = await FindEquivalentSubsription(subscription);
+                if (equivalentSubscription != null)
+                {
+                    return BadRequest(
```

Comment:
This should probably return the "Conflict" status code, because the request is conflicting with something in the DB.